daemon/containerd/image_pull.go

```diff
@@ -48,17 +49,37 @@ func (i *ImageService) PullImage(ctx context.Context, image, tagOrDigest string,
 	resolver, _ := i.newResolverFromAuthConfig(ctx, authConfig)
 	opts = append(opts, containerd.WithResolver(resolver))
 
+	old, err := i.resolveDescriptor(ctx, ref.String())
+	if err != nil && !errdefs.IsNotFound(err) {
+		return err
+	}
+	p := platforms.Default()
+	if platform != nil {
+		p = platforms.Only(*platform)
+	}
+
+	available, _, _, _, err := images.Check(ctx, i.client.ContentStore(), old, p)
```

Comment:
`available` doesn't mean that all content is available.

> // If available is true, the caller can assume that required represents the
> // complete set of content required for the image.
> //
> // missing will have the components that are part of required but not available
> // in the provider.

https://github.com/containerd/containerd/blob/5c37d3827b01bff9b67bc9b73a60cc35f4bfb96a/images/image.go#L281-L290


You have to check for `len(missing) == 0`.